### PR TITLE
feat: update bcs structs for versioned contracts

### PR DESCRIPTION
**Pull request by npty**
# Description

This PR updates the bcs structs definition for versioned contracts

## Todo
- [x] ITS
- [x] Gateway
- [x] Gas Service
- [x] Relayer Discovery
- [x] Squid

### CODE REVIEW

File: src/bcs.ts [hunk 1]
```diff
@@ -198,14 +213,21 @@ function getITSStructs() {
         trusted_addresses: bcs.vector(bcs.string()),
     });
 
-    const ITS = bcs.struct('ITS', {
-        id: UID,
+    const ITSV0 = bcs.struct('ITSV0', {
```
Comment (milapsheth) [inline src/bcs.ts:216]:
```suggestion
    const ITSv0 = bcs.struct('ITS_v0', {
```
this was renamed

File: src/bcs.ts [hunk 2]
```diff
@@ -198,14 +213,21 @@ function getITSStructs() {
...
         registered_coin_types: Table,
         registered_coins: Bag,
+        relayer_discovery_id: bcs.Address,
+        version_control: VersionControl,
+    });
+
+    const ITS = bcs.struct('ITS', {
+        id: UID,
+        name: bcs.u64(),
+        value: ITSV0,
```
Comment (milapsheth) [inline src/bcs.ts:230]:
```suggestion
        value: ITSv0,
```

**Review by milapsheth — APPROVED** [review]

File: src/bcs.ts [hunk 3]
```diff
@@ -224,6 +231,26 @@ function getSquidStructs() {
     };
 }
 
+function getRelayerDiscoveryStructs() {
+    const { Table } = getCommonStructs();
+    const { VersionControl } = getVersionControlStructs();
+
+    const RelayerDiscoveryV0 = bcs.struct('RelayerDiscovery_v0', {
```
Comment (milapsheth) [inline src/bcs.ts:238]: I think RelayerDiscoveryV0 hasn't been renamed to this yet. Can you fix that?

Comment (milapsheth) [inline src/bcs.ts:238]:
```suggestion
    const RelayerDiscoveryv0 = bcs.struct('RelayerDiscovery_v0', {
```

File: src/bcs.ts [hunk 4]
```diff
@@ -224,6 +231,26 @@ function getSquidStructs() {
...
+        configurations: Table,
+        version_control: VersionControl,
+    });
+
+    const RelayerDiscovery = bcs.struct('RelayerDiscovery', {
+        id: UID,
+        name: bcs.U64,
+        value: RelayerDiscoveryV0,
```
Comment (milapsheth) [inline src/bcs.ts:246]:
```suggestion
        value: RelayerDiscoveryv0,
```

File: test/bcs.js [hunk 5]
```diff
@@ -91,9 +93,22 @@ describe.only('BCS', () => {
...
+        checkIdAndSize(relayerDiscoveryV0.value.configurations, '5339d11ffc9ae10e448b36b776533e1f08c646ad0441c7a0d410b1e0e5d28e58', '1');
+        expect(relayerDiscoveryV0.value.version_control.allowed_functions[0].contents)
+            .to.be.an('array')
+            .with.lengthOf(3)
+            .that.includes('register_transaction', 'remove_transaction', 'get_transaction');
+    });
 });
 
 // This function is used by getting the test data in bytes from the object id
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (milapsheth) [inline test/bcs.js:111]: why is this needed?

File: test/bcs.js [hunk 6]
```diff
@@ -91,9 +93,22 @@ describe.only('BCS', () => {
...
             .that.includes('pay_gas', 'add_gas', 'collect_gas', 'refund');
     });
+
+    it('should decode RelayerDiscoveryV0 object successfully', async () => {
+        const relayerDiscoveryV0 = bcsStructs.relayerDiscovery.RelayerDiscovery.parse(fromHEX(hexData.RelayerDiscoveryV0));
+
+        expect(relayerDiscoveryV0.id).to.equal('5dcab278dc93438e0705fc32023808927e09a29b1ae52eef6cb33b9250d9b871');
+        expect(relayerDiscoveryV0.name).to.equal('0');
+        checkIdAndSize(relayerDiscoveryV0.value.configurations, '5339d11ffc9ae10e448b36b776533e1f08c646ad0441c7a0d410b1e0e5d28e58', '1');
+        expect(relayerDiscoveryV0.value.version_control.allowed_functions[0].contents)
```
Comment (milapsheth) [inline test/bcs.js:103]: instead of hardcoding allowed function names, use the goldenTest helper to dump these values. Makes it easier to auto update them and review

File: test/bcs.js (hunk 5 above)
Comment (npty) [reply test/bcs.js:111]: this function is used for obtaining the bcs bytes when adding new decoding tests. i don't want to remove this function, but the linter failed because it's unused. so i added this line so that the linter will skip the check here.

File: test/bcs.js (hunk 6 above)
Comment (npty) [reply test/bcs.js]: could you point me to the goldenTest helper function and how to use it to dump these values?

Comment (milapsheth) [reply test/bcs.js]: See our public interface [test](https://github.com/axelarnetwork/axelar-cgp-sui/blob/0fbd985733790607095b7cd94b37db5b895886e0/test/packages.js#L50)
